### DISCUSSION TITLE
what's the difference between "~18" and "18.*" when configuring BP_NODE_VERSION

### DISCUSSION THREAD

**Discussion by zhoufenqin** [paketo-buildpacks/feedback#62] [General]
I see some buildpack use "<version>.*" to set the version, like set `BP_GO_VERSION=1.19.*`

In nodejs buildpack, set node version to 18, I can use `BP_NODE_VERSION=18.*` or `BP_NODE_VERSION=~18`, does it the same?

**Comment by robdimsdale** [comment]
@zhoufenqin generally yes. All of the buildpacks will parse valid [semantic versions](https://semver.org/).

In addition, many (possibly all, I'm not sure) buildpacks use https://github.com/Masterminds/semver to parse and compare the versions. That library has broader support than the semver specification above. More details on what it supports can be found in the [README of that library](https://github.com/Masterminds/semver/blob/master/README.md). In particular, the sections on [`-` ranges](https://github.com/Masterminds/semver#hyphen-range-comparisons), [`*` comparisons](https://github.com/Masterminds/semver#wildcards-in-comparisons), [`~` ranges](https://github.com/Masterminds/semver#tilde-range-comparisons-major) and [`^` ranges](https://github.com/Masterminds/semver#caret-range-comparisons-major) are useful.
